packages/vitest/src/utils/retry.ts

```diff
@@ -0,0 +1,70 @@
+import type { SerializedConfig } from '../runtime/config'
+
+export function normalizeRetryConfig(
+  retry: SerializedConfig['retry']
+): {
+  count: number
+  strategy: 'immediate' | 'test-file' | 'deferred'
+  delay: number
+  condition?: string | RegExp | ((error: Error) => boolean)
+} {
+  if (typeof retry === 'number') {
+    return {
+      count: retry,
+      strategy: 'immediate',
+      delay: 0,
+    }
+  }
+
+  return {
+    count: retry.count ?? 0,
+    strategy: retry.strategy ?? 'immediate',
+    delay: retry.delay ?? 0,
+    condition: retry.condition,
+  }
+}
+
+export function shouldRetryOnError(
+  error: Error,
+  condition?: string | RegExp | ((error: Error) => boolean)
+): boolean {
+  if (!condition) {
+    return true
+  }
+
+  if (typeof condition === 'string') {
+    return error.message.includes(condition)
+  }
+
+  if (condition instanceof RegExp) {
+    return condition.test(error.message)
+  }
+
+  if (typeof condition === 'function') {
+    try {
+      return condition(error)
+    } catch {
+      return false
+    }
+  }
+
+  return true
+}
+
+export function sleep(ms: number): Promise<void> {
+  return new Promise(resolve => setTimeout(resolve, ms))
+}
+
+export function calculateBackoffDelay(
+  baseDelay: number,
+  retryCount: number,
+  algorithm: 'linear' | 'exponential' = 'linear'
```

Comment:
I am a bit confused why this needs to be customizable when we always pass down `exponential`. Should this check be removed?